e2e_playwright/st_altair_chart_basic_select.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import altair as alt
+import pandas as pd
+from altair.expr import datum
+from vega_datasets import data
+
+import streamlit as st
+
+# SCATTER CHART
+st.header("Altair Chart with point and interval selection")
+cars = data.cars()
+
+interval = alt.selection_interval()
+
+point = alt.selection_point()
+
+base = (
+    alt.Chart(cars)
+    .mark_point()
+    .encode(
+        x="Horsepower:Q",
+        y="Miles_per_Gallon:Q",
+        color=alt.condition(point, "Origin:N", alt.value("lightgray")),
+        tooltip=["Name", "Origin", "Horsepower", "Miles_per_Gallon"],
+    )
+)
+
+chart_point = base.add_params(point)
+st.subheader("Scatter chart with selection_point")
+st.altair_chart(chart_point, on_select=True, key="scatter_point")
+if st.session_state.scatter_point:
```

Comment:
This is slightly weird but it feels good enough to just check if there is a return value and something stored in session state for the right key.